kubernetes-client/src/main/java/io/fabric8/kubernetes/client/GracePeriodConfigurable.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client;
+
+public interface GracePeriodConfigurable<T>
+{
+    T withGracePeriodSeconds(Long gracePeriodSeconds);
```

Comment:
From @sarvigalava:

> The parameter in DeleteOptions model has the type Long, so I reused that. (as well as the name of the parameter 'gracePeriodSeconds')

OK let's leave it as a Long, although I don't really like that tbh - would prefer it to be an int, but generated type is a Long due to source type being go's int64.

Definitely rename though.